File: src/main/java/com/coffeebara/cafe/vo/CafeNoteResponse.java
package com.coffeebara.cafe.vo;

import java.util.List;

import com.coffeebara.media.vo.MediaAttachmentResponse;

public record CafeNoteResponse(
	Long id,
	String kakaoPlaceId,
	String title,
	String noteText,
	Integer displayOrder,
	List<MediaAttachmentResponse> attachments,
	String createdAt,
	String updatedAt
) {
}

File: frontend/app/lib/server/cafe-note-response.js
function toNumber(value) {
  if (value == null) {
    return null;
  }

  return Number(value);
}

function toStringValue(value) {
  if (value == null) {
    return "";
  }

  return String(value);
}

function toIsoString(value) {
  if (!(value instanceof Date)) {
    return "";
  }

  return value.toISOString();
}

export function toCafeNoteResponse(record, attachments = []) {
  return {
    id: toNumber(record?.id),
    kakaoPlaceId: toStringValue(record?.kakaoPlaceId),
    title: toStringValue(record?.note?.title),
    noteText: toStringValue(record?.note?.noteText),
    displayOrder: toNumber(record?.displayOrder) ?? 0,
    attachments,
    createdAt: toIsoString(record?.createdAt),
    updatedAt: toIsoString(record?.updatedAt),
  };
}
